Classes/Services/Document/DocumentManager.php

```diff
@@ -323,12 +329,34 @@ protected function updateRemotely($document, $workflowTransition = null, $delete
         }
 
         if ($this->getDocumentTransferManager()->update($document)) {
-            $this->removeDocument($document);
+
+            if(!$this->checkEmbargoDate($document)){
+                $this->removeDocument($document);
+            } else {
+                $document->setState(DocumentWorkflow::LOCAL_STATE_IN_PROGRESS . ':' . $document->getRemoteState());
+            }
             return $document->getDocumentIdentifier();
         }
 
         return false;
     }
 
+    /**
+     * @param $document
+     * @return bool (true: if no embargo is set or embargo is expired, false: embargo is active)
+     * @throws \Exception
+     */
+    protected function checkEmbargoDate($document)
```

Comment:
Naming: `Check embargo` could mean anything. This method tells if there is an active embargo. So it should be named `hasActiveEmbargo` accordingly.

Reply:
Renamed